### DISCUSSION TITLE
metadata in include-in-header?

### DISCUSSION THREAD

**Discussion by giabaio** [quarto-dev/quarto-cli#4069] [Q&A]
Hi all,
As usual, I may be missing the obvious trick, but is it possible to pass an element of the metadata or `params` to the `include-in-header` block, when rendering to pdf?

As a simple example, I'd like to do the following
```
---
title: "My title here"
params: 
  level: "Level 7"
format: 
  pdf:
    classoption:
     - twoside
include-in-header: 
      text: |
        \usepackage{fancyhdr}
        \pagestyle{fancy}
        \fancyhead[L]{\textit{STAT0019 --- {{< params.level >}} }}
        \fancyhead[R]{\textit{Page \thepage}}
        \fancyfoot[C]{}
        \fancyfoot[RE]{\largeContinued}
        \fancyfoot[RO]{\large Turn Over}
        \AtEndDocument{\fancyfoot[R]{\large End of Paper}}
        \usepackage{hyperref}
        \hypersetup{colorlinks=true,linkcolor=blue,filecolor=magenta,urlcolor=blue}
---

# Some chapter
Here the text
Blah Blah Blah
```
and my intention would be to write the variable `params.level` onto the header (as defined by `fancyhdr`). The problem is that doing the above simply renders the `{{< params.level >}}` verbatim.  I also know you could probably use different, more `koma`-friendly packages to write the header, but everything else actually works... So I don't think it's a specific `fancyhdr` issue.

Should I define a `LaTeX` variable something like `\def\level{...}`? But then again, what should I feed the `tex` definition that would be read from the `include-in-header` block?

Thanks for your help!

**Comment by mcanouil** [comment]
Did you try to use an actual shortcode, especially the meta shortcode? 🤔 I do not know if it works but has definitely more chances than without, i.e., `{{< params.level >}}` is not a shortcode thus Quarto has no reason to do something with it, 
`{{< meta params.level >}}`, on the other hand, is a shortcode defined by Quarto to read YAML metadata.

**Comment by giabaio** [comment]
I did, actually... Passing `{{< meta params.level >}}` is parsed as `< meta params.level >`. I figured `LaTeX` may mess up with curly brackets, but even doing `\{\{< meta params.level >\}\}` simply parse the string **literally** and is rendered as `{{< meta params.level >}}`...

**Reply by baptiste** [reply]
Your example works for me, after I correct the typo in `\largeContinued` which throws LaTeX off (and with `{{< meta params.level >}}`). 
Actually, I'd first assumed you'd need pandoc templating variables, as in `$params.level$`, but that doesn't work. The order in which things are processed (and by which tool in the chain) is always a bit of a mystery, I wonder if there's an in-depth documentation for this.

**Comment by giabaio** [comment]
Yes ---sorry. There was a typo in the `\large` command. Does the header get parsed as "STAT0019 - Level 7" for you?

**Reply by baptiste** [reply]
It does. I'm using a fairly recent version of quarto (built from source last week), maybe it makes a difference.

**Comment by giabaio** [comment]
I've pulled from here on Sunday, I think, so that should be OK... Not quite sure what's happening...

**Comment by baptiste** [comment]
For reference here's the full example I'm using

```
---
title: "My title here"
params: 
  level: "Level 7"
format: 
  pdf:
    classoption:
     - twoside
include-in-header: 
      text: |
        \usepackage{lipsum}
        \usepackage{fancyhdr}
        \pagestyle{fancy}
        \fancyhead[L]{\textit{STAT0019 --- {{< meta params.level >}} }}
        \fancyhead[R]{\textit{Page \thepage}}
        \fancyfoot[C]{}
        \fancyfoot[RE]{\large Continued}
        \fancyfoot[RO]{\large Turn Over}
        \AtEndDocument{\fancyfoot[R]{\large End of Paper}}
        \usepackage{hyperref}
        \hypersetup{colorlinks=true,linkcolor=blue,filecolor=magenta,urlcolor=blue}
---

\lipsum

```

<img width="1436" alt="Screenshot 2023-01-24 at 08 34 32" src="https://user-images.githubusercontent.com/18970/214132861-82024879-1f8d-4d36-9f0d-c22d4818f42e.png">

**Comment by giabaio** [comment]
This is *really* weird... I've copied/pasted your code (removing the `lipsum` bit) and all worked OK!... :+1: 

Thank you --- I'm still nonethewiser, but I'll take the win... :-)

**Reply by baptiste** [reply]
if you have the file under version control, the diff might show a small typo / extra space or something of the sort
